pkg/tsdb/elasticsearch/elasticsearch.go

```diff
@@ -2,50 +2,111 @@ package elasticsearch
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/datasource"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/instancemgmt"
+	"github.com/grafana/grafana/pkg/components/simplejson"
 	"github.com/grafana/grafana/pkg/infra/httpclient"
-	"github.com/grafana/grafana/pkg/models"
-	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin"
+	"github.com/grafana/grafana/pkg/plugins/backendplugin/coreplugin"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/tsdb"
 	es "github.com/grafana/grafana/pkg/tsdb/elasticsearch/client"
-	"github.com/grafana/grafana/pkg/tsdb/interval"
 )
 
-// ElasticsearchExecutor represents a handler for handling elasticsearch datasource request
-type Executor struct {
-	httpClientProvider httpclient.Provider
-	intervalCalculator interval.Calculator
+var eslog = log.New("tsdb.elasticsearch")
+
+func init() {
+	registry.Register(&registry.Descriptor{
+		Name:         "ElasticSearchService",
+		InitPriority: registry.Low,
+		Instance:     &Service{},
+	})
+}
+
+type Service struct {
+	BackendPluginManager backendplugin.Manager `inject:""`
+	Cfg                  *setting.Cfg          `inject:""`
```

Comment:
Don't think this is in use by this package?

```suggestion
```

Reply:
Probably not needed at all, yes.